ui/src/components/WebRTCVideo.tsx

```diff
@@ -144,27 +169,39 @@ export default function WebRTCVideo() {
     return () => {
       abortController.abort();
     };
-  }, [isPointerLockPossible, videoElm]);
+  }, [isPointerLockPossible]);
 
   const requestFullscreen = useCallback(async () => {
-    videoElm.current?.requestFullscreen({
+     if (!isFullscreenEnabled || !videoElm.current) return;
```

Comment:
This is hugely important... we need to request both the keyboard lock and pointer lock _before_ we transition to full-screen or things are just not stable.